src/foundry/foundry.js/clientDocuments/chatMessage.d.ts

```diff
@@ -1,16 +1,219 @@
+import { BaseToken, BaseUser } from '../../common/documents.mjs';
+
 declare global {
   /**
    * The client-side ChatMessage document which extends the common BaseChatMessage abstraction.
    * Each ChatMessage document contains ChatMessageData which defines its data schema.
    *
-   * @see {@link data.ChatMessageData}              The ChatMessage data sche
-   * ma
+   * @see {@link data.ChatMessageData}              The ChatMessage data schema
    * @see {@link documents.Messages}                The world-level collection of ChatMessage documents
    *
    * @param data - Initial data provided to construct the ChatMessage document
    */
-  class ChatMessage extends ClientDocumentMixin(foundry.documents.BaseChatMessage) {}
+  class ChatMessage extends ClientDocumentMixin(foundry.documents.BaseChatMessage) {
+    constructor(
+      data?: ConstructorParameters<ConstructorOf<foundry.documents.BaseChatMessage>>[0],
+      context?: ConstructorParameters<ConstructorOf<foundry.documents.BaseChatMessage>>[1]
+    );
+
+    /**
+     * If the chat message contains a Roll instance, cache it here
+     * @defaultValue `null`
+     */
+    protected _roll: Roll | null;
+
+    /**
+     * Return the recommended String alias for this message.
+     * The alias could be a Token name in the case of in-character messages or dice rolls.
+     * Alternatively it could be a User name in the case of OOC chat or whispers.
+     */
+    get alias(): string;
+
+    /**
+     * Is the current User the author of this message?
+     */
+    get isAuthor(): boolean;
+
+    /**
+     * Return whether the content of the message is visible to the current user.
+     * For certain dice rolls, for example, the message itself may be visible while the content of that message is not.
+     */
+    get isContentVisible(): boolean;
+
+    /**
+     * Test whether the chat message contains a dice roll
+     */
+    get isRoll(): boolean;
+
+    /**
+     * Return the Roll instance contained in this chat message, if one is present
+     */
+    get roll(): Roll | null;
+
+    /**
+     * Return whether the ChatMessage is visible to the current User.
+     * Messages may not be visible if they are private whispers.
+     */
+    get visible(): boolean;
+
+    /**
+     * The User who created the chat message.
+     */
+    get user(): User | undefined;
+
+    /** @override */
+    prepareData(): void;
+
+    /**
+     * Transform a provided object of ChatMessage data by applying a certain rollMode to the data object.
+     * @param chatData - The object of ChatMessage data prior to applying a rollMode preference
+     * @param rollMode - The rollMode preference to apply to this message data
+     * @returns The modified ChatMessage data with rollMode preferences applied
+     */
+    static applyRollMode(
+      chatData: foundry.data.ChatMessageData,
+      rollMode: ValueOf<typeof CONST.DICE_ROLL_MODES>
```

Comment:
The `ValueOf` use here and all others should be replaced with `CONST.DiceRollMode`.